apps/desktop/src/lib/select/Select.svelte

```diff
@@ -55,6 +55,10 @@
 	let listOpen = $state(false);
 	let inputBoundingRect = $state<DOMRect>();
 
+	$effect(() => {
+		filteredOptions = options;
+	});
+
```

Comment:
@Caleb-T-Owens @ndom91 Not sure if this is actually anti-pattern in Svelte or not.
What do you guys think?

Reply:
Yeah so ideally we'd not use $effect to sync state, they recommend using a callback if available. 

Check out this video if you're curious haha - https://youtu.be/7N4maEDhy4w

Reply:
> Yeah so ideally we'd not use $effect to sync state,
>

Yeah that makes sense.

 Thanks for the video!

> they recommend using a callback if available.
>

A callback :) ?

> Would a $derived do the trick here?
>

I expected that but it seems like `$derived` variables are not re-assignable.
I also expected that `$state` variables would be re-set to the prop that initialized it, but seem like not :/

Reply:
Oh wait, I think the issue is somewhere deeper.

Reply:
Talked to @ndom91, and there seems to be no other way to handling this. We agreed to leave it like that for the time being